src/Support/ResolveInfoFieldsAndArguments.php

```diff
@@ -132,15 +143,67 @@ private function foldSelectionSet(SelectionSetNode $selectionSet, int $descend):
         return $fields;
     }
 
-    private function getValue(ArgumentNode $argumentNode)
+    /**
+     * @param ValueNode $value
+     * @return mixed
+     * @throws \Exception
+     */
+    private function getValue(ValueNode $value)
     {
-        $value = $argumentNode->value;
         if ($value instanceof VariableNode) {
             $variableName = $value->name->value;
 
             return $this->info->variableValues[$variableName] ?? null;
         }
+        // Scalar Types
+        if ($value instanceof IntValueNode || $value instanceof FloatValueNode || $value instanceof StringValueNode || $value instanceof BooleanValueNode || $value instanceof EnumValueNode) {
+            return $value->value;
+        }
+        // null Type
+        if ($value instanceof NullValueNode) {
+            return;
+        }
+        // object Type
+        if ($value instanceof ObjectValueNode) {
+            return $this->getInputObjectValue($value);
+        }
+        // list object Type
+        if ($value instanceof ListValueNode) {
+            return $this->getInputListObjectValue($value);
+        }
+
+        throw new \Exception('Failed to resolve unknown ValueNode type');
+    }
+
+    /**
+     * @param ObjectValueNode $objectValueNode
+     * @return array
+     * @throws \Exception
+     */
```

Comment:
Remove the docblock please (throws should not be used and the rest does not add anything)